### PR TITLE
SEAB-4704: delete entries that are not considered any of the EntryTypes

### PR DESCRIPTION
**Pull request by hyunnaye**
**Description**
Prior to this PR, the SQL query `select count(userid) from user_entry where userid = <redacted>;` would return a higher number than the endpoint
```
curl -s -X 'GET'   'http://localhost:4200/api/users/users/entries'   -H 'accept: application/json'   -H 'Authorization: Bearer <redacted>' | jq '. | length'
```

Upon investigation, the extra entries found in `select count(userid) from user_entry where userid = <redacted>;` are entries that are essentially non-existent where it is not considered an apptool, tool, workflow, or service.

We can get every entry ids of with this issue by :
```
select * from user_entry ue where ue.entryid not in  (select id from workflow union all
select id from apptool union all
select id from service union all
select id from tool );
```

Thus this PR deletes all of these entries from the database leading to an output of 0 when running the query above again.

**Review Instructions**
Verify that the query is correct and deletes all of these entries.

**Issue**
https://ucsc-cgl.atlassian.net/browse/SEAB-4704

**Security**
If there are any concerns that require extra attention from the security team, highlight them here.

Please make sure that you've checked the following before submitting your pull request. Thanks!

- [x] Check that you pass the basic style checks and unit tests by running `mvn clean install`
- [x] Ensure that the PR targets the correct branch. Check the milestone or fix version of the ticket.
- [x] Follow the existing JPA patterns for queries, using named parameters, to avoid SQL injection
- [x] If you are changing dependencies, check the Snyk status check or the dashboard to ensure you are not introducing new high/critical vulnerabilities
- [x] Assume that inputs to the API can be malicious, and sanitize and/or check for Denial of Service type values, e.g., massive sizes
- [x] Do not serve user-uploaded binary images through the Dockstore API
- [x] Ensure that endpoints that only allow privileged access enforce that with the `@RolesAllowed` annotation
- [x] Do not create cookies, although this may change in the future
- [x] If this PR is for a user-facing feature, create and link a documentation ticket for this feature (usually in the same milestone as the linked issue). Style points if you create a documentation PR directly and link that instead.

### CODE REVIEW

**Review by coverbeck — APPROVED** [review]
For other reviewers, see Nayeon's comment [here](https://ucsc-cgl.atlassian.net/browse/SEAB-4704?focusedCommentId=44353) -- it looks like no other orphaned user_entry rows have been created since March 2020. The orphaning somehow got fixed since then, so it seems like we don't need to worry about this recurring.

File: dockstore-webservice/src/main/resources/migrations.1.14.0.xml [hunk 1]
```diff
@@ -276,4 +276,12 @@
         <addUniqueConstraint columnNames="metricsid" constraintName="uk_metricsid_version_metrics" tableName="version_metrics"/>
         <addForeignKeyConstraint baseColumnNames="metricsid" baseTableName="version_metrics" constraintName="fk_metricsid_version_metrics" deferrable="false" initiallyDeferred="false" onDelete="NO ACTION" onUpdate="NO ACTION" referencedColumnNames="id" referencedTableName="metrics"/>
     </changeSet>
+    <changeSet id="deleteNonExistentEntries" author="nhyun">
+        <sql dbms="postgresql">
+            DELETE FROM user_entry ue WHERE ue.entryid NOT IN (SELECT id FROM workflow UNION ALL
+                                                               SELECT id FROM apptool UNION ALL
+                                                               SELECT id from service UNION ALL
```
Comment (svonworl) [inline dockstore-webservice/src/main/resources/migrations.1.14.0.xml:283]: Probably should add a SELECT from the `notebook` table to UNION, just to foreclose the possibility of any problems if the `notebook` table somehow has rows.

**Review by svonworl — APPROVED** [review]

**Review by denis-yuen — APPROVED** [review]
neat!
 (also sorry for this inheritance pattern that breaks foreign key constraints, sigh)

File: dockstore-webservice/src/main/resources/migrations.1.14.0.xml [hunk 2]
```diff
@@ -276,4 +276,13 @@
...
```
Comment (kathy-t) [inline dockstore-webservice/src/main/resources/migrations.1.14.0.xml:283]: Nit, `from` should be capitalized

**Review by kathy-t — APPROVED** [review]

**Comment by svonworl** [comment]
> neat! (also sorry for this inheritance pattern that breaks foreign key constraints, sigh)

Was wondering, would it be possible to create a "view" table that aggregates the entry IDs from the different subentity tables, and use that for any desired foreign key constraints?

**Comment by denis-yuen** [comment]
> > neat! (also sorry for this inheritance pattern that breaks foreign key constraints, sigh)
> 
> Was wondering, would it be possible to create a "view" table that aggregates the entry IDs from the different subentity tables, and use that for any desired foreign key constraints?

I don't know, but we have this pattern in a couple places, so I'd love for us to find out. 
I think @NatalieEO has a stored procedure or something in one place, but a constraint on a materialised table or something could be more intuitive (if a little un-likely)

**Pull request merged**